helix-core/src/main/java/org/apache/helix/task/WorkflowConfig.java

```diff
@@ -552,6 +559,11 @@ public Builder setConfigMap(Map<String, String> cfg) {
         }
       }
 
+      if (cfg.containsKey(ZNRecord.ENABLE_COMPRESSION_BOOLEAN_FIELD)) {
```

Comment:
No need to check containsKey, unless the default value enableCompression is true.